hollow/src/main/java/com/netflix/hollow/api/consumer/HollowConsumer.java

```diff
@@ -958,11 +1005,58 @@ public B withGeneratedAPIClass(Class<? extends HollowAPI> generatedAPIClass) {
             return (B)this;
         }
 
+        /**
+         * Specifies a filter config. Clears any type filter specified with {@code withTypeFilter(...)}.
+         *
+         * @return this builder
+         * @see #withTypeFilter(TypeFilter)
+         * @see #withTypeFilter(UnaryOperator)
+         */
         public B withFilterConfig(HollowFilterConfig filterConfig) {
-            this.filterConfig = filterConfig;
+            /*
+             * TODO: deprecate this once TypeFilter's field filtering implemented
+             */
+            this.typeFilter = filterConfig;
             return (B)this;
         }
 
+        /**
+         * Specifies the type filter. Clears any filter config specified with {@code withFilterConfig()}
+         * @param typeFilter
+         * @return this builder
+         * @see #withTypeFilter(UnaryOperator)
+         * @see #withFilterConfig(HollowFilterConfig)
+         */
+        public B withTypeFilter(TypeFilter typeFilter) {
```

Comment:
I wonder if it would be beneficial to document the limitations of `TypeFilter` and `HollowFilterConfig` in javadoc for `withTypeFilter` and `withHollowFilterConfig`. Depending on the time frame for `TypeFilter` feature completeness and when we can drop `HollowFilterConfig`, this might help avoid support cost for questions like "which one should I use?". Ofcourse we'd have to keep the `withTypeFilter` javadoc up to date as we develop it, or else the failure mode is that a feature is ready but not used.

Reply:
I have field level filtering working, so I can deprecate `HollowFilterConfig` now.

👍 to documenting the limitations in the JavaDoc. `TypeFilter` now has only the 1 limitation: has no serialization/deserialization like `HollowFilterConfig`.

Reply:
Done'd.